uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-widget/leave-widget.component.js

```diff
@@ -183,15 +210,15 @@ define([
     }
 
     /**
-     * Loads all absence periods and stores the current one.
+     * Loads all absence periods and stores the current one or the last absence
+     * period.
      *
      * @return {Promise}
      */
-    function loadCurrentAbsencePeriod () {
+    function loadAbsencePeriod () {
       return AbsencePeriod.all().then(function (periods) {
-        vm.currentAbsencePeriod = _.find(periods, function (period) {
-          return period.current;
-        });
+        vm.absencePeriod = findCurrentAbsencePeriod(periods) ||
```

Comment:
i think it would be better if you had something like this
```js
vm.absencePeriod = findAbsencePeriodToDisplay();

function findAbsencePeriodToDisplay () {
  return currentAbsencePeriod() || lastAbsencePeriod();
}
```